setlexsem/generate/generate_sets.py

```diff
@@ -45,19 +49,23 @@ def make_sets_from_sampler(
     num_runs: int,
 ) -> List[Dict[str, Any]]:
     """Generate random sets from the sampler"""
-    # create the dataset
+
+    # initlize the dataset
     set_list = []
     for i in range(num_runs):
-        # create two sets from the sampler
-        A, B = sample_set()
-        # loop through operations (on the same random sets)
-        set_list.append(
-            {
-                "experiment_run": i,
-                "A": A,
-                "B": B,
-            }
-        )
+        try:
```

Comment:
Do we know when does sample_set() throws an exception apart from when it's the base class instance. We can add a unit test as well for failure when sampling.

Reply:
I like this, can you propose a unit test for this? It fails when the generator is empty. If there's a way to check for generator output without emptying it out, let's use that logic.